### ISSUE TITLE
MeshPhysicalMaterial: Shader does not properly recompile when sheenTint changes

### ISSUE THREAD

**Issue by WestLangley** [mrdoob/three.js#22465]
A shader recompile is required when `.sheenTint` changes from black to not-black, or vice versa.

Since `sheenTint` is a `Color`, it appears that getting this to work is going to require something more than a simple setter/getter. 

Colors are not typically assigned, they are `.set()`.

 - Three.js version: [r132, r133dev]

**Comment by WestLangley** [comment]
I hope there is a reasonable solution short of adding `._onChangedCallback()` to the `Color` class.

Suggestions welcome.

**Comment by mrdoob** [comment]
I think I would do what I mentioned in https://github.com/mrdoob/three.js/pull/22381#issuecomment-904827044.

Even if the gltf spec doesn't have it, we can add a `sheen` float property that multiplies the color and reuses the same code from `clearcoat` and `transmission` for triggering program compiles.

**Comment by WestLangley** [comment]
tldr; -- there is none, sorry

//

That is not a bad idea, but here is the reason not to -- and I expect this is the same reason it was removed from the updated Dassault spec, and not included in the glTF spec.

When `Color` is used for a light color, an intensity is _required_ because `intensity` is where the units come from -- _lux_, for example. The color is a tint, or mask, and is unit-less, or dimensionless.

When `Color` is used for a material color or for a sheen tint, intensity is not required. A material color is a _reflectance_, which is dimensionless. A sheen tint is also a reflectance and is dimensionless.

Adding sheen intensity, a dimensionless quantity -- and restricted to [ 0, 1 ], no less -- adds nothing. It is not needed.

I assume this is why there is no sheen intensity parameter in the glTF or Dassault models.

//

Also, `sheen = 0` is the same as `sheen = 1` with `sheenTint = black`. So we still have the same problem to solve.

@mrdoob ... but whatever you prefer is OK with me.

**Comment by WestLangley** [comment]
tldr; -- there is none, sorry

//

That is not a bad idea, but here is the reason not to -- and I expect this is the same reason it was removed from the updated Dassault spec, and not included in the glTF spec.

When `Color` is used for a light color, an intensity is _required_ because `intensity` is where the units come from -- _lux_, for example. The color is a tint, or mask, and is unit-less, or dimensionless.

When `Color` is used for a material color or for a sheen tint, intensity is not required. A material color is a _reflectance_, which is dimensionless. A sheen tint is also a reflectance and is dimensionless.

Adding sheen intensity, a dimensionless quantity -- and restricted to [ 0, 1 ], no less -- adds nothing. It is not needed.

I assume this is why there is no sheen intensity parameter in the glTF or Dassault models.

//

I'll file a PR, anyway.
